src/test/mir-opt/matches_reduce_branches.foo.MatchBranchSimplification.32bit.diff

```diff
@@ -4,38 +4,23 @@
   fn foo(_1: Option<()>) -> () {
       debug bar => _1;                     // in scope 0 at $DIR/matches_reduce_branches.rs:6:8: 6:11
       let mut _0: ();                      // return place in scope 0 at $DIR/matches_reduce_branches.rs:6:25: 6:25
-      let mut _2: bool;                    // in scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
-      let mut _3: isize;                   // in scope 0 at $DIR/matches_reduce_branches.rs:7:22: 7:26
+      let mut _2: isize;                   // in scope 0 at $DIR/matches_reduce_branches.rs:7:22: 7:26
   
       bb0: {
-          StorageLive(_2);                 // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
-          _3 = discriminant(_1);           // scope 0 at $DIR/matches_reduce_branches.rs:7:22: 7:26
--         switchInt(move _3) -> [0_isize: bb2, otherwise: bb1]; // scope 0 at $DIR/matches_reduce_branches.rs:7:22: 7:26
-+         _2 = Eq(_3, const 0_isize);      // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
-+         goto -> bb3;                     // scope 0 at $DIR/matches_reduce_branches.rs:7:22: 7:26
+          _2 = discriminant(_1);           // scope 0 at $DIR/matches_reduce_branches.rs:7:22: 7:26
+          switchInt(move _2) -> [0_isize: bb2, otherwise: bb1]; // scope 0 at $DIR/matches_reduce_branches.rs:7:22: 7:26
```

Comment:
lol, ok so we have two competing optimizations now. if the one from this PR triggers, the other won't trigger anymore. 

I think we should wait for https://github.com/rust-lang/rust/pull/78191 to get merged and then see how to adjust or duplicate the tests to actually show what is being changed.

Reply:
I'm leaning towards simply deleting the `foo` function from `matches_reduce_branches` now that ConstGoto does the simplification transformation

Reply:
Oh I'm not worried about who does what, but about the fact that this code doesn't optimize as well anymore as it did.

The switch now basically decides between a match arm that assigns to `_0` and one that doesn't. Otherwise the match arms are the same. So after `SimplifyLocals`, these arms are the same. Maybe move the function to its own test file and ensure that the final MIR is just a single basic block with a `return` statement?